buildSrc/src/main/java/net/neoforged/neodev/NeoDevPlugin.java

```diff
@@ -0,0 +1,527 @@
+            generateBinPatchesTask.configure(task -> {
+                task.setGroup(INTERNAL_GROUP);
+                task.classpath(binpatcherConfig);
+                task.getPatchedJar().set(tasks.named("jar", Jar.class).flatMap(Jar::getArchiveFile));
+                task.getSourcePatchesFolder().set(sourcesPatchesFolder);
+                task.getMappings().set(createCleanArtifacts.flatMap(CreateCleanArtifacts::getMergedMappings));
+            });
+        }
+
+        return new BinaryPatchOutputs(
+                generateMergedBinPatches.flatMap(GenerateBinaryPatches::getOutputFile),
+                generateClientBinPatches.flatMap(GenerateBinaryPatches::getOutputFile),
+                generateServerBinPatches.flatMap(GenerateBinaryPatches::getOutputFile)
+        );
+    }
+
+    private record BinaryPatchOutputs(
+            Provider<RegularFile> binaryPatchesForMerged,
+            Provider<RegularFile> binaryPatchesForClient,
+            Provider<RegularFile> binaryPatchesForServer
+    ) {
+    }
+
+    /**
+     * Sets up NFRT, and creates the sources and resources artifacts.
+     */
+    static TaskProvider<CreateMinecraftArtifacts> configureMinecraftDecompilation(Project project) {
+        project.getPlugins().apply(NeoFormRuntimePlugin.class);
+
+        var configurations = project.getConfigurations();
+        var dependencyFactory = project.getDependencyFactory();
+        var tasks = project.getTasks();
+        var neoDevBuildDir = project.getLayout().getBuildDirectory().dir("neodev");
+
+        var rawNeoFormVersion = project.getProviders().gradleProperty("neoform_version");
+        var minecraftVersion = project.getProviders().gradleProperty("minecraft_version");
+        var mcAndNeoFormVersion = minecraftVersion.zip(rawNeoFormVersion, (mc, nf) -> mc + "-" + nf);
+
+        // Configuration for all artifacts that should be passed to NFRT to prevent repeated downloads
+        var neoFormRuntimeArtifactManifestNeoForm = configurations.create("neoFormRuntimeArtifactManifestNeoForm", spec -> {
+            spec.setCanBeConsumed(false);
+            spec.setCanBeResolved(true);
+            spec.getDependencies().addLater(mcAndNeoFormVersion.map(version -> {
+                return dependencyFactory.create("net.neoforged:neoform:" + version);
+            }));
+        });
+
+        tasks.withType(NeoFormRuntimeTask.class, task -> {
+            task.addArtifactsToManifest(neoFormRuntimeArtifactManifestNeoForm);
+        });
+
+        return tasks.register("createSourceArtifacts", CreateMinecraftArtifacts.class, task -> {
+            var minecraftArtifactsDir = neoDevBuildDir.map(dir -> dir.dir("artifacts"));
+            task.getSourcesArtifact().set(minecraftArtifactsDir.map(dir -> dir.file("base-sources.jar")));
+            task.getResourcesArtifact().set(minecraftArtifactsDir.map(dir -> dir.file("minecraft-local-resources-aka-client-extra.jar")));
```

Comment:
There shouldn't be a reason for the stupidly long file name in NeoDev, I would just leave it at `minecraft-resources.jar`

Reply:
It actually has to contain `client-extra` for the BSL ignoreList to ignore it.

Reply:
BSL filters are not `contains`, they're `startsWith`

Reply:
The only effect `client-extra` would have is when being found by the dev provider (see https://github.com/neoforged/FancyModLoader/blob/496b519a95d604fbf361dd14a6185e8a883e1540/loader/src/main/java/net/neoforged/fml/loading/moddiscovery/locators/NeoForgeDevProvider.java#L40-L51). But the latest version of FML (which current neo and subsequent neo will use) also has a classpath fallback, so it should not be needed.

Reply:
Done! :)